[#299] Update changelog (main) (#304)
irods/irods_capability_automated_ingest: CHANGELOG.md

# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
and this project **only** adheres to the following _(as defined at [Semantic Versioning](https://semver.org/spec/v2.0.0.html))_:

> Given a version number MAJOR.MINOR.PATCH, increment the:
> 
> - MAJOR version when you make incompatible API changes
> - MINOR version when you add functionality in a backward compatible manner

> > - PATCH version when you make backward compatible bug fixes

Reviewer: I would argue that we will only start adhering to this post-1.0. Thoughts? Is that even worth mentioning? This is a breaking release in the sense that it's a new Celery app with a different name.

Reviewer 2: > This is a breaking release in the sense that it's a new Celery app with a different name.

is that true?

Reviewer 2: breaking in what sense...?

Reviewer: It's a new celery app: https://github.com/irods/irods_capability_automated_ingest/commit/1c2ec9d638fcb77e30107559432258348d49f3e1

Note the README and test changes where workers are given a different application name

Reviewer 2: yes, a new app - but not breaking in that anyone who's using it... may use that old name anywhere?

what scenario / use case would anyone notice?

Reviewer: I think everything besides the name of the app is the same. The old app name is gone, so that cannot be used anymore: https://github.com/irods/irods_capability_automated_ingest/commit/b598d9bd01ba0b61dfbdd1cfeac2dcfeaff702a8

Reviewer 2: got it.

we're pre 1.0.0.

i think we're good without any additional words.

Reviewer: Okay. Resolving

> ## [0.6.0] - 2024-10-08

Author: Date must match the release date.

Author: Changed to 2024-10-11

> ## [0.6.0] - 2024-10-08

Author: Seems the version is supposed to be hyperlinked, although I don't think it matters much.

Thoughts?

Author: FYI - Seems other projects which follow the changelog rules do not link the version number to the commit/range/tag.

> This release adds the ability to delete data objects and collections from iRODS which DO NOT exist in the source tree. The feature exposes two new events and an event handler method for controlling the delete mode.

Reviewer 2: ```suggestion
This release adds the ability to delete data objects and collections from iRODS which DO NOT exist in the source being ingested. The feature exposes two new events and an event handler method for controlling the delete mode.
```

Author: Done

### Added

- Add Delete mode (48, 261, 288).
- Track start time of sync jobs (92).
- Track jobs stopped by "irods_sync stop" (210).
- Add Docker Compose project for testing (244).

### Changed

- Improve documentation (23, 137, 140, 150, 183, 214, 257, 289).

> - list subcommand displays more information about jobs (91).

Reviewer 2: ```suggestion
- The `list` subcommand displays more information about jobs (91).
```

Author: Updated the "watch" subcommand reference too.

Done.

> - `list` subcommand displays more information about jobs (91).

Reviewer 2: this is the only item that does not start with a verb.

Author: Updated

- Improve testing, Docker demo, and code formatting (100, 132, 180).
- Replace Celery application and tasks (211).

> - Remove /tmp mount directory from Docker test harness (235).

Author: L31 can be included with L29.

Thoughts?

Reviewer 2: fine to keep separate

Author: Leaving as is.

- Bump PRC dependency version to <3.0.0 (263).

> - Code refactor and clean up (180, 272, 274).

Reviewer 2: uses an action word to start the line...

```suggestion
- Refactor and clean up code (180, 272, 274).
```

Author: Done

- Improve handling of irods_session (269).
- Disable mingling for Celery works in tests (280).
- Replace use of "sync" with "tasks" in Celery tasks names (281).

> ### Removed

Reviewer 2: are the sections always ordered in this way?  should/can `Removed` be after `Fixed`?

Author: Not sure. Will check. I used the ordering from the S3 API changelog.

Author: There's a conversation about the order of the sections in the keep-a-changelog repo.

See https://github.com/olivierlacan/keep-a-changelog/issues/458.

It's not enforced currently, but people feel it should. I think as long as we're consistent, we're fine. I do like the idea of following the same rules as everyone else though.

There's an alternative ordering included in that conversation too.

Reviewer 2: i defer to whatever feels right to you.   choosing the default / what others are doing is always defensible.

Author: Reading that conversation about orderings again, I think the proposed ordering from that conversation is the right answer. Going to use that.
- Security
- Changed
- Removed
- Deprecated
- Fixed
- Added

Will update the S3 API changelog as well.

Author: Done

Reviewer 2: 'Added' being last definitely feels weird.   But I get that it's 'in order of importance' for someone concerned about upgrades and compatibility.

---

Just for posterity... I think my default (today) would be...

 - Security
 - Added (Features/Enhancements)
 - Fixed (Bugs)
 - Changed (Refactors)
 - Deprecated
 - Removed
